contracts/DutchXBaseModule.sol

```diff
@@ -0,0 +1,151 @@
+pragma solidity ^0.5.0;
+import "@gnosis.pm/safe-contracts/contracts/base/Module.sol";
+
+
+/// @title DutchX Base Module - Expose a set of methods to enable a Safe to interact with a DX
+/// @author Denis Granha - <denis@gnosis.pm>
+contract DutchXBaseModule is Module {
+
+    address public dutchXAddress;
+    // isWhitelistedToken mapping maps destination address to boolean.
+    mapping (address => bool) public isWhitelistedToken;
+    mapping (address => bool) public isOperator;
+
+    // helper variables used by the CLI
+    address[] public whitelistedTokens; 
+    address[] public whitelistedOperators;
+
+    /// @dev Setup function sets initial storage of contract.
+    /// @param dx DutchX Proxy Address.
+    /// @param tokens List of whitelisted tokens.
+    /// @param operators List of addresses that can operate the module.
+    /// @param _manager Address of the manager, the safe contract.
+    function setup(address dx, address[] memory tokens, address[] memory operators, address payable _manager)
+        public
+    {
+        require(address(manager) == address(0), "Manager has already been set");
+        if (_manager == address(0)){
+            manager = ModuleManager(msg.sender);
+        }
+        else{
+            manager = ModuleManager(_manager);
+        }
+
+        dutchXAddress = dx;
+
+        for (uint256 i = 0; i < tokens.length; i++) {
+            address token = tokens[i];
+            require(token != address(0), "Invalid token provided");
```

Comment:
should we check `isWhitelistedToken[token] == false` to avoid duplicate tokens in the array?

Reply:
true, we can add it as a patch in a later commit. For now is up to the user to use it well